### DISCUSSION TITLE
How to make an argument optional?

### DISCUSSION THREAD

**Discussion by tippfelher** [MichalLytek/type-graphql#1437] [Q&A]
My code
```
  @Query(returns => [Post], { description: "" })
  async posts(
    @Arg("order") order?: PostOrder
  ): Promise<Post[]> {
  }
```

Problem: Argument is not made optional. Schema generated:

```
type Query {
  posts(order: PostOrder!): [Post!]!
}
```

Expected: order can be undefined

Not a solution: Marking order as nullable because null is obviously not undefined.

**Comment by MichalLytek** [comment]
Have you tried `{ nullable: true }`, just like with fields, queries and mutations?

https://github.com/MichalLytek/type-graphql/blob/master/examples/enums-and-unions/resolver.ts#L17

**Reply by tippfelher** [reply]
I haven't tried it before I asked my question, because nullable is not what I want.

However, I did try it now and it does exactly what I expected. With your method,

order is of type PostOrder | null | undefined

but I want

order is of type PostOrder |  undefined

Thanks

**Reply by tippfelher** [reply]
I just realized that this may be a graphql limitation and I can't do anything against it.
